Add libhttpserver dependency

skaupper opened the pull request:
We need the `libhttpserver` library for the REST server(s). There are no real packages for the package managers out there. How would you prefer to integrate it? Below I added an approach using a install script which clones the repo as a submodule and builds and installs the lib from there.

Thoughts?

wurmmi commented:
we should maybe add the additional installation step for libhttpserver somewhere in the documentation.

other than that - LGTM :slightly_smiling_face:

skaupper commented:
I added libmicrohttpd as well, since it is a dependency of libhttpserver.
Install scripts and CMake detections are included as well as a section in the README about the installation process (which is pretty much just calling `sudo scripts/install_libmicrohttpd.sh` and `sudo scripts/install_libhttpserver.sh`).

I added an examples directory to be able to add small applications for testing or prototype purposes (in this case I tested if the example from the libhttpserver compiles and links without errors).

Please test building and installing dependencies as well as the examples, so we can merge this PR.

janeson332 commented on examples/hello_libhttpserver.cpp:
```diff
@@ -0,0 +1,30 @@
+ *
+ * This program starts a server on port 8080 which listens to the endpoint /hello
+ * To verify that it's working use `url -XGET %http://localhost:8080/hello` or use
+ * a browser and go to <http://localhost:8080/hello>
```
typo, think you meaned `curl` ^^ not `url`
think also the % is wrong

skaupper replied:
true, it should indeed be `curl`
the % is used to prevent automatic link generation, so that it wont get a clickable link
